spec/spec_helper.rb

```diff
@@ -1,116 +1,74 @@
     end
 
-    # These two settings work together to allow you to limit a spec run
-    # to individual examples or groups you care about by tagging them with
-    # `:focus` metadata. When nothing is tagged with `:focus`, all examples
-    # get run.
     config.filter_run :focus
     config.run_all_when_everything_filtered = true
-
-    # Allows RSpec to persist some state between runs in order to support
-    # the `--only-failures` and `--next-failure` CLI options. We recommend
-    # you configure your source control system to ignore this file.
     config.example_status_persistence_file_path = 'spec/.rspec-state'
-
-    # Limits the available syntax to the non-monkey patched syntax that is
-    # recommended. For more details, see:
-    #   - http://myronmars.to/n/dev-blog/2012/06/rspecs-new-expectation-syntax
-    #   - http://www.teaisaweso.me/blog/2013/05/27/rspecs-new-message-expectation-syntax/
-    #   - http://myronmars.to/n/dev-blog/2014/05/notable-changes-in-rspec-3#new__config_option_to_disable_rspeccore_monkey_patching
     config.disable_monkey_patching!
-
-    # This setting enables warnings. It's recommended, but in some cases may
-    # be too noisy due to issues in dependencies.
     config.warnings = true
 
-    # Many RSpec users commonly either run the entire suite or an individual
-    # file, and it's useful to allow more verbose output when running an
-    # individual spec file.
-    if config.files_to_run.one?
-      # Use the documentation formatter for detailed output,
-      # unless a formatter has already been configured
-      # (e.g. via a command-line flag).
-      config.default_formatter = 'doc'
-    end
+    config.default_formatter = 'doc' if config.files_to_run.one?
 
-    # Print the 10 slowest examples and example groups at the
-    # end of the spec run, to help surface which specs are running
-    # particularly slow.
     config.profile_examples = 10
 
-    # Run specs in random order to surface order dependencies. If you find an
-    # order dependency and want to debug it, you can fix the order by providing
-    # the seed, which is printed after each run.
-    #     --seed 1234
     config.order = :random
-
-    # Seed global randomization in this process using the `--seed` CLI option.
-    # Setting this allows you to use `--seed` to deterministically reproduce
-    # test failures related to randomization by passing the same `--seed` value
-    # as the one that triggered the failure.
     Kernel.srand config.seed
+
+    # Add VCR to all tests
+    config.around(:each) do |example|
+      options = example.metadata[:vcr] || { record: (ENV['VCR_RECORD_MODE'] ? ENV['VCR_RECORD_MODE'].to_sym : :none) }
+      if options[:record] == :skip
+        VCR.turned_off(&example)
+      else
+        name = example.metadata[:full_description].split(/\s+/, 2).join('/').underscore.gsub(/\./, '/').gsub(/[^\w\/]+/, '_').gsub(/\/$/, '')
+        VCR.use_cassette(name, options, &example)
+      end
+    end
```

Comment:
I added the separate `spec.rb` file in order to separate out some of our settings and behavior from the spec_helper.  I don't have a strong opinion about doing it that way, but I think it makes it more clear how specs are set up since you then end up using cleaned up methods and well named methods as opposed to inlined code that isn't as clear.  I think that would make sense here since there is a lot going on in these few lines and they affect every spec in the suite.